modules/builder/src/builders/helpers/xviz-trajectory-helper.js

```diff
@@ -132,8 +133,16 @@ export function getObjectTrajectory({
  * @returns {Array} Vector [x, y] in meters
  */
 export function getGeospatialVector(from, to, heading = 0) {
-  const fromCoord = turf.point([from.longitude, from.latitude]);
-  const toCoord = turf.point([to.longitude, to.latitude]);
+  const fromCoord = addMetersToLngLat(
+    [from.longitude, from.latitude, from.altitude],
+    [from.x || 0, from.y || 0, from.z || 0]
+  );
+
+  const toCoord = addMetersToLngLat(
```

Comment:
`addMetersToLngLat` uses approximation. Since you're already importing turfjs here, use [turf.destination](http://turfjs.org/docs/#destination)